test_conformance/basic/test_readimage3d.cpp

```diff
@@ -113,120 +111,114 @@ prepare_reference(unsigned char * input_ptr, int w, int h, int d)
-  if (err)
-    return -1;
-
-  cl_sampler sampler = clCreateSampler(context, CL_FALSE, CL_ADDRESS_CLAMP_TO_EDGE, CL_FILTER_NEAREST, &err);
-  test_error(err, "clCreateSampler failed");
-
-  err  = clSetKernelArg(kernel[0], 0, sizeof streams[0], &streams[0]);
-  err |= clSetKernelArg(kernel[0], 1, sizeof streams[2], &streams[2]);
-  err |= clSetKernelArg(kernel[0], 2, sizeof sampler, &sampler);
-  test_error(err, "clSetKernelArg failed");
-
-  err  = clSetKernelArg(kernel[1], 0, sizeof streams[1], &streams[1]);
-  err |= clSetKernelArg(kernel[1], 1, sizeof streams[2], &streams[2]);
-  err |= clSetKernelArg(kernel[1], 2, sizeof sampler, &sampler);
-  test_error(err, "clSetKernelArg failed");
-
-    threads[0] = (unsigned int)img_width;
-    threads[1] = (unsigned int)img_height;
-     threads[2] = (unsigned int)img_depth;
-
-  for (i=0; i<2; i++)
-  {
-    err = clEnqueueNDRangeKernel(queue, kernel[i], 3, NULL, threads, NULL, 0, NULL, NULL);
-    test_error(err, "clEnqueueNDRangeKernel failed");
-
-    err = clEnqueueReadBuffer(queue, streams[2], CL_TRUE, 0, length, output_ptr, 0, NULL, NULL);
-    test_error(err, "clEnqueueReadBuffer failed");
-
-    switch (i)
-    {
-      case 0:
-        err = verify_3d_image8(ref_ptr[i], output_ptr, img_width, img_height, img_depth);
-        if ( err != 0 )
-            log_info("READ_IMAGE3D_BGRA_UNORM_INT8 test passed\n");
-        break;
-      case 1:
-        err = verify_3d_image8(ref_ptr[i], output_ptr, img_width, img_height, img_depth);
-        if ( err != 0 )
-            log_info("READ_IMAGE3D_RGBA_UNORM_INT8 test passed\n");
-        break;
-    }
-
-    if (err)
-      break;
-  }
-
-    // cleanup
-  clReleaseSampler(sampler);
-    clReleaseMemObject(streams[0]);
-    clReleaseMemObject(streams[1]);
-    clReleaseMemObject(streams[2]);
-  for (i=0; i<2; i++)
-  {
-    clReleaseKernel(kernel[i]);
-    clReleaseProgram(program[i]);
-  }
-    free(input_ptr[0]);
-    free(input_ptr[1]);
-    free(output_ptr);
-  free(ref_ptr[0]);
-  free(ref_ptr[1]);
-
-    return err;
+	cl_mem streams[2];
+	cl_program program;
+	cl_kernel kernel;
+	cl_sampler sampler;
+	static cl_image_format img_format[2] =
+	{
+		{CL_BGRA, CL_UNORM_INT8},
+		{CL_RGBA, CL_UNORM_INT8},
+	};
+	static const char* kernelName[] =
+	{
+		"test_bgra8888",
+		"test_rgba8888",
+	};
+	static const char* kernelSourceString[] =
+	{
+		bgra8888_kernel_code,
+		rgba8888_kernel_code,
+	};
+	unsigned char *input_ptr;
+	float *output_ptr;
+	double *ref_ptr;
+	size_t threads[3];
+	int img_width = 64;
+	int img_height = 64;
+	int img_depth = 64;
+	int i, err;
+	size_t origin[3] = {0, 0, 0};
+	size_t region[3] = {img_width, img_height, img_depth};
+	size_t length = img_width * img_height * img_depth * 4 * sizeof(float);
+
+	PASSIVE_REQUIRE_3D_IMAGE_SUPPORT( device )
+
+	for (uint32_t i = 0; i < 2; i++)
+	{
+		if (is_image_format_required(img_format[i], CL_MEM_READ_ONLY, CL_MEM_OBJECT_IMAGE3D, device))
```

Comment:
You could reverse the logic and use `continue` to reduce the indentation of the entire block below.